### ISSUE TITLE
Piscina and @types/node > 20.8.0

### ISSUE THREAD

**Issue by ilteoood** [jasnell/piscina#426]
We are experiencing some issues with @types/node > 20.8.0 and piscina.
Piscina depends on the module `eventemitter-asyncresource` which has not been updated within 3 years now.

If I try to upgrade to @types/node version 20.8.6, during the build phase this error is raised:
```
Error: node_modules/eventemitter-asyncresource/dist/src/index.d.ts(20,15): error TS2417: Class static side 'typeof EventEmitterAsyncResource' incorrectly extends base class static side 'typeof EventEmitter'.
  Types of property 'EventEmitterAsyncResource' are incompatible.
    Types of construct signatures are incompatible.
      Type 'new (options?: string | Options) => EventEmitterAsyncResource' is not assignable to type 'new (options?: EventEmitterAsyncResourceOptions) => EventEmitterAsyncResource'.
        Construct signature return types 'EventEmitterAsyncResource' and 'EventEmitterAsyncResource' are incompatible.
          The types of 'asyncId' are incompatible between these types.
            Type '() => number' is not assignable to type 'number'.
Error: Process completed with exit code 2.
```

I'm using `typescript@5.1.6` and this is my tsconfig:
```
{
  "compilerOptions": {
    "module": "commonjs",
    "esModuleInterop": true,
    "allowJs": true,
    "allowSyntheticDefaultImports": true,
    "target": "es2017",
    "noImplicitAny": true,
    "moduleResolution": "node16",
    "resolveJsonModule": true,
    "sourceMap": true,
    "outDir": "dist"
  },
  "include": [
    "src/**/*"
  ]
}
```

I hope these informations are enough, otherwise I'll try to create a minimal repro later today.

**Comment by metcoder95** [comment]
Yeah, that's totally valid; now that `EventEmitterAsyncResource` is in core we might want to migrate and remove the dependency at all beyond only TS

**Comment by garybernhardt** [comment]
It looks like a fix is in progress in #433. (Thanks @groozin!)

However, just one note to add, in case it matters: this also happens with @types/node 18.18.6. It errors in exactly the way that @ilteoood showed above.
